src/Grok.Net.Tests/UnitTests.cs

```diff
@@ -73,15 +78,47 @@ public void TimePatternTest()
         public void EmailPatternTest()
         {
             Grok act = new Grok("%{EMAILADDRESS:email}:%{GREEDYDATA:comment}");
-            string logs = @"Bob.Davis@microsoft.com:Free as in Free Beer";
+            string emailAddress ="Bob.Davis@microsoft.com";
+            string comment = "Free as in Free Beer";
+            string logs = emailAddress + ":" + comment;
             
             GrokResult grokResult = act.Parse(logs);
            
-            Assert.Equal("Bob.Davis@microsoft.com", grokResult[0].Value);
-            Assert.Equal("Free as in Free Beer", grokResult[1].Value);
+            Assert.Equal(emailAddress, grokResult[0].Value);
+            Assert.Equal(comment, grokResult[1].Value);
             Assert.Equal("email", grokResult[0].Key);
             Assert.Equal("comment", grokResult[1].Key);
-
         }            
+
+        [Theory]
+        [MemberData(nameof(GetIpv4PatternTestData))]
+        public void IPv4PatternTest(string grokPattern, string logs, string ipAddress, string comment, int grokElementOffset)
+        {
+            Grok act = new Grok(grokPattern);
+            
+            GrokResult grokResult = act.Parse(logs);
+           
+            Assert.Equal(ipAddress, grokResult[grokElementOffset].Value);
+            Assert.Equal(comment, grokResult[grokElementOffset+1].Value);
+        }  
+        public static IEnumerable<object[]> GetIpv4PatternTestData()
```

Comment:
Could you please explain why we really need this for only one test? What is the value?

Reply:
You seemed keen on re-use, using a test data injector allows for injection of additional test cases to the test at a later date. You could argue that all tests are only ever one test, with different test data applied to them.

Reply:
this named `GetIpv4PatternTestData`, so, it can be used only in one test (test of IP). I don't think it makes any sense until we have only one test for IP. Yes, create some data that will be used over test is okay, but we have one test for IP and use it for the one test `IPv4PatternTest`

> You could argue that all tests are only ever one test, with different test data applied to them.

 don't understand this. could you please explain?